cockatrice/src/userinfobox.cpp

```diff
@@ -140,3 +162,126 @@ void UserInfoBox::processResponse(const Response &r)
+
+    QString email = QString::fromStdString(user.email());
+    int gender = user.gender();
+    QString country = QString::fromStdString(user.country());
+    QString realName = QString::fromStdString(user.real_name());
+
+    DlgEditUser dlg(this, email, gender, country, realName);
+    if(!dlg.exec())
+        return;
+
+    Command_AccountEdit cmd;
+    cmd.set_real_name(dlg.getRealName().toStdString());
+    cmd.set_email(dlg.getEmail().toStdString());
+    cmd.set_gender((ServerInfo_User_Gender) dlg.getGender());
+    cmd.set_country(dlg.getCountry().toStdString());
+    
+    PendingCommand *pend = client->prepareSessionCommand(cmd);
+    connect(pend, SIGNAL(finished(Response, CommandContainer, QVariant)), this, SLOT(processEditResponse(const Response &)));
+    
+    client->sendCommand(pend);
+}
+
+void UserInfoBox::actPassword()
+{
+    DlgEditPassword dlg(this);
+    if(!dlg.exec())
+        return;
+
+    Command_AccountPassword cmd;
+    cmd.set_old_password(dlg.getOldPassword().toStdString());
+    cmd.set_new_password(dlg.getNewPassword().toStdString());
+    
+    PendingCommand *pend = client->prepareSessionCommand(cmd);
+    connect(pend, SIGNAL(finished(Response, CommandContainer, QVariant)), this, SLOT(processPasswordResponse(const Response &)));
+    
+    client->sendCommand(pend);
+}
+
+void UserInfoBox::actAvatar()
+{
+    DlgEditAvatar dlg(this);
+    if(!dlg.exec())
+        return;
+
+    Command_AccountImage cmd;
+    cmd.set_image(dlg.getImage().data(), dlg.getImage().size());
+    
+    PendingCommand *pend = client->prepareSessionCommand(cmd);
+    connect(pend, SIGNAL(finished(Response, CommandContainer, QVariant)), this, SLOT(processAvatarResponse(const Response &)));
+    
+    client->sendCommand(pend);
+}
+
+void UserInfoBox::processEditResponse(const Response &r)
+{
+    switch (r.response_code()) {
+        case Response::RespOk:
+            updateInfo(nameLabel.text());
+            QMessageBox::information(this, tr("Information"), tr("User information updated."));
+            break;
+        case Response::RespFunctionNotAllowed:
+            QMessageBox::critical(this, tr("Error"), tr("This server does not permit you to update your user informations."));
+            break;
+        case Response::RespInternalError:
+        default:
+            QMessageBox::critical(this, tr("Error"), tr("An error occured while trying to update your user informations."));
+            break;
+    }
+}
+
+void UserInfoBox::processPasswordResponse(const Response &r)
+{
+    switch (r.response_code()) {
+        case Response::RespOk:
+            QMessageBox::information(this, tr("Information"), tr("Password changed."));
+            break;
+        case Response::RespFunctionNotAllowed:
+            QMessageBox::critical(this, tr("Error"), tr("This server does not permit you to change your password."));
+            break;
+        case Response::RespPasswordTooShort:
```

Comment:
We are not validating the username here, only the new password. The only (hardcoded) rule by now is length >= 6 chars